sdk/core/azure-core/src/http/curl/curl_session_private.hpp

```diff
@@ -418,6 +424,13 @@ namespace Azure { namespace Core { namespace Http {
      * @return The size of the payload.
      */
     int64_t Length() const override { return m_contentLength; }
+
+    /**
+     * @brief Return the network connection if the server indicated that the connection is upgraded.
+     *
+     * @return The network connection, or null if the connection was not upgraded.
+     */
+    std::unique_ptr<CurlNetworkConnection> ExtractConnection();
```

Comment:
nit on naming:
```suggestion
    std::unique_ptr<CurlNetworkConnection> ExtractUpgradedConnection();
```

Reply:
You're not extracting an upgraded connection. There is no such thing as an "upgraded connection" to the `CurlNetworkConnection` (but there is such a thing as a "connection").

An upgraded connection is a construct in the WebSockets protocol, not the HTTP protocol.